### PR TITLE
feat: add hyperband series

### PR DESCRIPTION
**Pull request by be-marc**

### CODE REVIEW

File: mlr-org/gallery/series/2022-12-02-hyperband-subsampling/index.qmd [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+lgr::get_logger("mlr3")$set_threshold("warn")
+lgr::get_logger("bbotk")$set_threshold("warn")
+
+set.seed(2)
+```
+
+# Scope
+
+We continue working with the *Hyperband* optimization algorithm [@li_2018].
+The [previous post](/gallery/series/2022-12-01-hyperband-xgboost) used the number of boosting iterations of an XGBoost model as the resource.
```
Comment (sebffischer) [inline mlr-org/gallery/series/2022-12-02-hyperband-subsampling/index.qmd:32]:
```suggestion
The [previous post](/gallery/series/2022-12-01-hyperband-xgboost) used the number of boosting iterations of an XGBoost model as the fidelity parameter, here also referred to as the resource.
```

File: mlr-org/gallery/series/2022-12-02-hyperband-subsampling/index.qmd [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+However, Hyperband is not limited to machine learning algorithms that are trained iteratively.
+The resource can also be the number of features, the training time of a model, or the size of the training data set.
+In this post, we will tune a support vector machine and use the size of the training data set as the fidelity parameter.
+The time to train a support vector machine and the performance increases with the size of the data set.
+This makes the data set size a suitable fidelity parameter for Hyperband.
+This is the second part of the Hyperband series.
+The first part can be found here [Hyperband Series - Iterative Training](/gallery/series/2022-12-01-hyperband-xgboost).
```
Comment (sebffischer) [inline mlr-org/gallery/series/2022-12-02-hyperband-subsampling/index.qmd:39]:
```suggestion
The first part can be found here [Hyperband Series (I) - Iterative Training](/gallery/series/2022-12-01-hyperband-xgboost).
```

File: mlr-org/gallery/series/2022-12-02-hyperband-subsampling/index.qmd [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+graph_learner$timeout = c(train = 30, predict = 30)
+graph_learner$fallback = lrn("classif.featureless")
+```
+
+Let's create the tuning instance.
+We use the `"none"` terminator because Hyperband controls the termination itself.
+
+```{r}
+instance = ti(
+  task = tsk("sonar"),
```
Comment (sebffischer) [inline mlr-org/gallery/series/2022-12-02-hyperband-subsampling/index.qmd:113]: why is the indentiation here different compared to the search space definition above?

File: mlr-org/gallery/series/2022-12-01-hyperband-xgboost/index.qmd [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+  subsample         = to_tune(1e-1, 1)
+)
+```
+
+We construct the tuning instance.
+We use the `"none"` terminator because Hyperband terminates itself when all brackets are evaluated.
+
+```{r}
+instance = ti(
+  task = tsk("spam"),
```
Comment (sebffischer) [inline mlr-org/gallery/series/2022-12-01-hyperband-xgboost/index.qmd:213]: indentiation sollte mit search space übereinstimmen

File: mlr-org/gallery/series/2022-12-02-hyperband-subsampling/index.qmd [hunk 5]
```diff
@@ -0,0 +1,214 @@
+---
+title: "Hyperband Series - Data Set Subsampling"
+description: |
+  Optimize the hyperparameters of a Support Vector Machine with Hyperband.
+categories:
+  - tuning
+  - classification
+author:
+  - name: Marc Becker
+    url: https://github.com/be-marc
```
Comment (sebffischer) [inline mlr-org/gallery/series/2022-12-02-hyperband-subsampling/index.qmd:10]:
```suggestion
    url: https://github.com/be-marc
  - name: Sebastian Fischer
    url: https://github.com/sebffischer
```

File: mlr-org/gallery-top-series.yml [hunk 6]
```diff
@@ -1,3 +1,7 @@
+- title: Hyperband Series
```
Comment (sebffischer) [inline mlr-org/gallery-top-series.yml:1]:
```suggestion
- title: Hyperband Series (II)
```

**Review by sebffischer — COMMENTED** [review]
Noch ein paar Kommentare: 

Also der große Unterschied zwischen iterative training und dem subsampling ist ja noch, dass man das iterative theoretisch hotstarten kann. Geht das nicht mit dem Algorithmus? 

Also das fände ich noch wichtig. 

Außerdem noch eine organisatorische Frage: Vielleicht wäre ein .bib file für die ganze gallery besser? Vieles wird sich da ja vielleicht schon überschneiden

File: mlr-org/gallery/series/2022-12-01-hyperband-xgboost/index.qmd [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+future::plan("multisession")
+
+set.seed(1)
+```
+
+# Scope
+
+Increasingly large data sets and search spaces make hyperparameter optimization a time-consuming task.
+*Hyperband* [@li_2018] solves this by approximating the performance of a configuration on a simplified version of the problem such as a small subset of the training data, with just a few training epochs in a neural network, or with only a small number of iterations in a gradient-boosting model.
+After sampling random configurations, Hyperband iteratively allocates more resources to promising configurations and terminates low-performing ones.
```
[the remark this answers is not in the document]
Comment (be-marc) [reply mlr-org/gallery/series/2022-12-01-hyperband-xgboost/index.qmd:36]: > Außerdem noch eine organisatorische Frage: Vielleicht wäre ein .bib file für die ganze gallery besser? Vieles wird sich da ja vielleicht schon überschneiden

Ja ich zitiere ständig das gleiche.

**Comment by be-marc** [comment]
> Also der große Unterschied zwischen iterative training und dem subsampling ist ja noch, dass man das iterative theoretisch hotstarten kann. Das geht auch mit der Implementierung oder?

Na ja also ich würde sagen der große Unterschied ist, dass iterative Algorithmen einen natürlichen Fidelity Parameter haben. Ohne Subsampling würden die anderen Algorithmen ja gar nicht gehen. Aber klar du hast recht, das ist ein Unterschied. Hotstarten funktioniert im Moment aber nicht effizient und ich rate in diesem [Post](https://github.com/mlr-org/mlr3website/pull/90/files) auch davon ab. Aber ich schreib noch eine kleine Anmerkung.

**Comment by be-marc** [comment]
> Außerdem noch eine organisatorische Frage: Vielleicht wäre ein .bib file für die ganze gallery besser? Vieles wird sich da ja vielleicht schon überschneiden

Gute Idee. Ich zitiere immer das selbe.
